src/views/Package/ChooseSubmodule/ChooseSubmodule.tsx

```diff
@@ -28,18 +23,28 @@ export const ChooseSubmodule: FunctionComponent<ChooseSubmoduleProps> = ({
 
   const [filter, setFilter] = useState("");
 
+  const packageName = scope ? `${scope}/${name}` : name;
+
   const onGoBack = () => {
-    const lang = query.get(QUERY_PARAMS.LANGUAGE);
-    push(`${pathname}${lang ? `?${QUERY_PARAMS.LANGUAGE}=${lang}` : ""}`);
+    push(
+      getPackagePath({
+        name: packageName,
+        version,
+        language,
+      })
+    );
```

Comment:
Previously we were preserving query params and hashes which caused unwanted behavior. If a user was viewing the docs for Construct Foo within submodule `Bar` and then switched docs to a different submodule or the main module, the page would try to render docs from the same path (Construct Foo), which would not exist